### DISCUSSION TITLE
Transferring .htaccess and .htpasswd Files to the Server

### DISCUSSION THREAD

**Discussion by gpsblues** [GetPublii/Publii#1798] [Help] [answered]
Hi! 

I’m currently working on a site with some password-protected pages using `.htaccess` and `.htpasswd`.

For deployment, I wanted to keep copies of these files in the root directory, so I placed them in **Tools/File Manager/root directory**.

However, I noticed that during FTP upload, only the `.htaccess` file gets transferred. I don’t believe the server is blocking `.htpasswd`, as I can upload it directly via an FTP client without any issues.

Is this a limitation or a security setting in Publii? 

Thanks a lot!

**Comment by dziudek** [comment]
@gpsblues - Publii has no such limitations during upload - do you have anything suspicious in your deployment logs regarding this file?

**Reply by bttrx** [reply, marked as answer]
@dziudek Same here.

Publii 0.46.2 (build 16966) ignores all files with a leading `.`, except `.htaccess`. These files are added to the `output` directory, but not to `files.publii.json`.

The offending line is:

https://github.com/GetPublii/Publii/blob/443986c3ab91eac3cc4e39901c044175543128df/app/back-end/modules/deploy/deployment.js#L557

`file.indexOf('.') !== 0` checks whether the filename does not start with a dot. This excludes hidden files.

**Reply by bttrx** [reply]
@dziudek Would you mind to add `.htpasswd` to the whitelist? (Maybe also needs changes at other places.)

**Reply by dziudek** [reply]
@bttrx - you're right - I have totally forgot about this fragment :) Thank you for spotting this - I will add a fix in v.0.46.3

**Reply by bttrx** [reply]
@dziudek Also reported as a bug in #1565.
